### PR TITLE
#246 Fix for inappropriate disposal

### PR DESCRIPTION
**Pull request by timyhac**
This Pull Request fixes inappropriate disposal (for tags that failed to initialize using InitializeAsync).

When InitializeAsync ReadAsync or WriteAsync times out, it also will ensure that it will fail with a ErrorTimeout as opposed to a ErrorAborted

### CODE REVIEW

File: src/Examples/CSharp NativeImport/CallbackEx.cs [hunk 1]
```diff
@@ -7,7 +7,12 @@ namespace NativeImport_Examples
 {
     public struct MyUserDataClass
     {
-        public int Value = 1234;
```
Comment (timyhac) [inline src/Examples/CSharp NativeImport/CallbackEx.cs:10]: This is unrelated to the primary change. Without it the code doesn't compile

**Comment by jkoplo** [comment]
Started to look at this, but I'm going to need to pull the whole code to think through how it interacts...

**Comment by jkoplo** [comment]
Alright went through it a little more. This makes sense, I think. I still haven't quite internalized how the whole ConcurrentStack is used with the native code and the TCS.

What's up with the broken build and the new GH action?

**Review by jkoplo — APPROVED** [review]

**Comment by timyhac** [comment]
I think it's this https://github.com/libplctag/libplctag.NET/issues/255#issue-1228493794
